extensions/cornerstone/src/Viewport/Overlays/CustomizableViewportOverlay.tsx

```diff
@@ -0,0 +1,532 @@
+   */
+  // Todo: this should act on the toolGroups instead of the toolbar state
+  useEffect(() => {
+    const { unsubscribe } = toolbarService.subscribe(
+      toolbarService.EVENTS.TOOL_BAR_STATE_MODIFIED,
+      () => {
+        setActiveTools(toolbarService.getActiveTools());
+      }
+    );
+
+    return () => {
+      unsubscribe();
+    };
+  }, [toolbarService]);
+
+  const _renderOverlayItem = useCallback(
+    item => {
+      const overlayItemProps: OverlayItemProps = {
+        element,
+        viewportData,
+        imageSliceData,
+        viewportIndex,
+        servicesManager,
+        customization: item,
+        formatters: {
+          formatPN: formatPN,
+          formatDate: formatDICOMDate,
+          formatTime: formatDICOMTime,
+          formatNumberPrecision: formatNumberPrecision,
+        },
+        instance: instance,
+        // calculated
+        voi,
+        scale,
+        instanceNumber,
+      };
+
+      if (item.customizationType === 'ohif.overlayItem.windowLevel') {
+        return <VOIOverlayItem {...overlayItemProps} />;
+      } else if (item.customizationType === 'ohif.overlayItem.zoomLevel') {
+        return <ZoomOverlayItem {...overlayItemProps} />;
+      } else if (item.customizationType === 'ohif.overlayItem.instanceNumber') {
+        return <InstanceNumberOverlayItem {...overlayItemProps} />;
+      } else {
+        const renderItem = customizationService.applyType(item);
+
+        if (typeof renderItem.content === 'function') {
+          return renderItem.content(overlayItemProps);
+        }
+      }
+    },
+    [
+      element,
+      viewportData,
+      imageSliceData,
+      viewportIndex,
+      servicesManager,
+      customizationService,
+      instance,
+      voi,
+      scale,
+      instanceNumber,
+    ]
+  );
+
+  const getTopLeftContent = useCallback(() => {
+    const items = topLeftCustomization?.items || [
+      {
+        id: 'WindowLevel',
+        customizationType: 'ohif.overlayItem.windowLevel',
+      },
+    ];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`topLeftOverlayItem_${i}`}>{_renderOverlayItem(item)}</div>
+        ))}
+      </>
+    );
+  }, [
+    topLeftCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
+    customizationService,
+    instance,
+    voi,
+    scale,
+    instanceNumber,
+  ]);
+
+  const getTopRightContent = useCallback(() => {
+    const items = topRightCustomization?.items || [
+      {
+        id: 'InstanceNmber',
+        customizationType: 'ohif.overlayItem.instanceNumber',
+      },
+    ];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`topRightOverlayItem_${i}`}>{_renderOverlayItem(item)}</div>
+        ))}
+      </>
+    );
+  }, [
+    topRightCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
```

Comment:
we don't depend on lots of these inside the `useCallback` why you are putting them here?